src/graph/validator/GoValidator.cpp

```diff
@@ -36,9 +36,23 @@ Status GoValidator::validateImpl() {
     return Status::SemanticError("$- must be referred in FROM before used in WHERE or YIELD");
   }
 
-  if (!exprProps.varProps().empty() && goCtx_->from.fromType != kVariable) {
-    return Status::SemanticError(
-        "A variable must be referred in FROM before used in WHERE or YIELD");
+  if (!exprProps.varProps().empty()) {
+    if (goCtx_->from.fromType != kVariable) {
+      return Status::SemanticError(
+          "A variable must be referred in FROM before used in WHERE or YIELD");
+    }
+    auto varPropsMap = exprProps.varProps();
+    std::vector<std::string> keys;
+    for (const auto& [k, v] : varPropsMap) {
+      keys.emplace_back(k);
+    }
+    if (keys.size() > 1) {
+      return Status::SemanticError("Multiple variable property is not supported in WHERE or YIELD");
+    }
+    if (keys.front() != goCtx_->from.userDefinedVarName) {
```

Comment:
This condition may never be reached

Reply:
Why?

Reply:
An error will be reported in the validatorFrom. when keys.front() != userdefinedVarName,

Reply:
I can debug this piece of code use this statement:
```
      $A = GO FROM 'Tim Duncan' OVER like YIELD like._dst AS dst;
       $B = GO FROM $A.dst OVER like YIELD like._dst AS dst;
        GO FROM $A.dst over like YIELD like._dst AS dst, $B.dst
```

Reply:
my fault, deduceProp do not work on from clause